### ISSUE TITLE
exclude_paths doesn't apply to yaml files

### ISSUE THREAD

**Issue by landure** [ansible/ansible-lint#4745]
##### Summary

`ansible-lint` report issues on a sops-encrypted yaml file, notably the first line that trigger the space requirement in comments:

```yaml
#ENC[…
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 25.7.0 using ansible-core:2.18.7 ansible-compat:25.6.0 ruamel-yaml:0.18.10 ruamel-yaml-clib:0.2.12
```

- ansible installation method: OS package (Nix OS unstable)
- ansible-lint installation method: OS package (Nix OS unstable)

##### STEPS TO REPRODUCE

Create a `secrets` path, with an invalid yaml file:

```bash
mkdir secrets
cat > 'secrets/test.yml' <<EOF
#comment without space after #
value: value
EOF
```

By convention, i store secrets in `secrets` path. `.ansible-lint.yml` contains:

```yaml
exclude_paths:
  - .cache/ # implicit unless exclude_paths is defined in config
  - .devenv/
  - .direnv/
  - "secrets/"
```

With this setting, `ansible-lint` still report issues on the yaml secrets files. Only by adding this to `.yamllint` does `ansible-lint` ignore the files:

```yaml
…
ignore:
  - secrets/
…
```

##### Desired Behavior

`ansible-lint` ignore yaml files in `ignored_paths`.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

`ansible-lint` lint yaml files in `ignored_paths`.
